src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+// Copyright 2015 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::cell::RefCell;
+use std::cmp::PartialEq;
+use std::default::Default;
+use std::ffi::CString;
+use std::io::{Error, ErrorKind};
+use std::marker::{Send, Sync};
+use std::mem;
+use std::ops::Deref;
+use std::env;
+use std::ptr;
+use std::slice;
+
+use libc::{intptr_t, memcmp};
+
+use uuid::Uuid;
+
+use serde;
+use bincode;
+use bincode::serde::DeserializeError;
+
+use winapi;
+use winapi::{HANDLE, INVALID_HANDLE_VALUE, LPVOID};
+use kernel32;
+
+// some debug bump macros to better track what's going on in case of errors
+lazy_static! {
+    static ref CURRENT_PROCESS_ID: winapi::ULONG = unsafe { kernel32::GetCurrentProcessId() };
+    static ref CURRENT_PROCESS_HANDLE: intptr_t = unsafe { kernel32::GetCurrentProcess() as intptr_t };
```

Comment:
Since it's a static, I don't see how `drop()` even matters here?...

And yes, I do realise why you can't use a raw `HANDLE` here -- but using `WinHandle` would avoid this problem, without needing a hacky unsafe workaround...